packages/toast/src/InternalToast/InternalToast.spec.tsx

```diff
@@ -0,0 +1,206 @@
+        const { queryByRole } = render(
+          <InternalToast title="hello world" variant={Variant.Success} />,
+        );
+
+        const progressBar = queryByRole('progressbar');
+        expect(progressBar).not.toBeInTheDocument();
+      });
+
+      test(`when 'progress' is set`, () => {
+        const { queryByRole } = render(
+          <InternalToast
+            title="hello world"
+            variant={Variant.Success}
+            progress={0}
+          />,
+        );
+
+        const progressBar = queryByRole('progressbar');
+        expect(progressBar).not.toBeInTheDocument();
+      });
+    });
+
+    describe('the correct icon is rendered', () => {
+      const expectedVariantIcons: Record<Variant, string> = {
+        [Variant.Success]: 'Checkmark With Circle Icon',
+        [Variant.Note]: 'Info With Circle Icon',
+        [Variant.Warning]: 'Warning Icon',
+        [Variant.Important]: 'Important With Circle Icon',
+        [Variant.Progress]: 'Refresh Icon',
+      };
+
+      test.each(Object.values(Variant) as Array<Variant>)(
+        `when 'variant' is '%s'`,
+        variant => {
+          const { getByLabelText } = render(
+            <InternalToast title="hello world" variant={variant} />,
+          );
+
+          expect(getByLabelText(expectedVariantIcons[variant])).toBeVisible();
+        },
+      );
+    });
+  });
+
+  describe('`title` prop', () => {
+    test('a string, the title element renders', () => {
+      const titleText = 'this is the title';
+      const { queryByText } = render(
+        <InternalToast title={titleText} variant={Variant.Success} />,
+      );
+
+      const body = queryByText(titleText);
+      expect(body).toBeVisible();
+    });
+
+    test('a JSX element, the title element renders', () => {
+      const titleText = 'this is the title';
+      const titleElement = <span>{titleText}</span>;
+      const { queryByText } = render(
+        <InternalToast title={titleElement} variant={Variant.Success} />,
+      );
+
+      const titleSpan = queryByText(titleText);
+      expect(titleSpan).toBeVisible();
+    });
+  });
+
+  describe('`description` prop', () => {
+    test('a string, the title element renders', () => {
+      const bodyText = 'this is the title';
+      const { queryByText } = render(
+        <InternalToast
+          title="hello"
+          description={bodyText}
+          variant={Variant.Success}
+        />,
+      );
+
+      const body = queryByText(bodyText);
+      expect(body).toBeVisible();
+    });
+
+    test('a JSX element, the body element renders', () => {
+      const bodyText = 'this is the body';
+      const bodyElement = <span>{bodyText}</span>;
+      const { queryByText } = render(
+        <InternalToast
+          title="hello"
+          description={bodyElement}
+          variant={Variant.Success}
+        />,
+      );
+
+      const bodySpan = queryByText(bodyText);
+      expect(bodySpan).toBeVisible();
+    });
```

Comment:
nit: maybe test by test-id to check that the span renders too, not just the body text